packages/compass-indexes/src/modules/index-view.ts

```diff
@@ -0,0 +1,31 @@
+import type { AnyAction } from 'redux';
+import { isAction } from '../utils/is-action';
+
+export type IndexView = 'regular-indexes' | 'search-indexes';
+
+export enum ActionTypes {
+  ChangeIndexView = 'index-list/ChangeIndexView',
+}
+
+type ChangeIndexViewAction = {
+  type: ActionTypes.ChangeIndexView;
+  view: IndexView;
+};
+
+export const INITIAL_STATE: IndexView = 'regular-indexes';
+
+export default function reducer(state = INITIAL_STATE, action: AnyAction) {
+  if (isAction<ChangeIndexViewAction>(action, ActionTypes.ChangeIndexView)) {
+    return action.view;
+  }
+
+  return state;
+}
+
+export const changeIndexView = (view: IndexView): ChangeIndexViewAction => ({
+  type: ActionTypes.ChangeIndexView,
+  view: view,
+});
+
+export const changedToRegularIndexes = () => changeIndexView('regular-indexes');
```

Comment:
nit: Not crazy about the naming, but not sure what's better. switchToRegularIndexes and switchToSearchIndexes? Just feels like this should be an action / verb in the present tense given that actions do things, they don't just notify that a thing happened.